governance/contract_connector.go

```diff
@@ -0,0 +1,155 @@
+package governance
+
+import (
+	"errors"
+	"math/big"
+	"strings"
+
+	"github.com/klaytn/klaytn/accounts/abi"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/blockchain/vm"
+	"github.com/klaytn/klaytn/common"
+	govcontract "github.com/klaytn/klaytn/contracts/gov"
+	"github.com/klaytn/klaytn/params"
+)
+
+type contractCaller struct {
+	chain        blockChain
+	contractAddr common.Address
+}
+
+var govParamAbi, _ = abi.JSON(strings.NewReader(govcontract.GovParamABI))
+
+const govparamfunc = "getAllParamsAt"
+
+func (c *contractCaller) getAllParamsAt(num *big.Int) (*params.GovParamSet, error) {
+	// respect error; some nodes can succeed without error while others do not
+	tx, evm, err := c.prepareCall(num)
+	if err != nil {
+		return nil, err
+	}
+
+	// ignore error; if error, all nodes will have the same error
+	res, _ := c.callTx(tx, evm)
+
+	// ignore error; if error, all nodes will have the same error
+	pset, err := c.parse(res)
+	if err != nil {
+		return params.NewGovParamSet(), nil
+	}
+	return pset, nil
+}
+
+func (c *contractCaller) prepareCall(num *big.Int) (*types.Transaction, *vm.EVM, error) {
```

Comment:
@ian0371 Please make this function more generic. `contractCaller` is intended to be a reusable module.
```suggestion
func (c *contractCaller) prepareCall(callAbi *abi.ABI, funcName string, num *big.Int) (*types.Transaction, *vm.EVM, error) {
```

Reply:
fixed in f8a0379